src/generate.c

```diff
@@ -292,6 +310,27 @@ int main(int argc, char** argv)
         FILE* f = fopen(generator_run_stamp, "w");
         g_assert(f != NULL);
         fclose(f);
+    } else if (check_called_just_in_time()) {
+        /* When booting with cloud-init, network configuration
+         * might be provided just-in-time. Specifically after
+         * system-generators were executed, but before
+         * network.target is started. In such case, auxiliary
```

Comment:
If this runs after network.target is started, does it matter?  If so, should the check_called_just_in_time() ensure it only returns true if network.target is not started?

Reply:
IMO this is correct. I introduced an additional check in `check_called_just_in_time()`.